reloj-checador/app/views/companies/index.html.erb

```diff
@@ -1,80 +1,78 @@
+              </td>
+            </tr>
+          <%end%>
+        </tbody>
+      </table>
+      <div class="d-flex justify-content-center align-items-center mt-5">
+        <%= pagy_bootstrap_nav(@pagy).html_safe %>
+      </div>
+    </div>
+  </div>
 </div>
 
 <%= render 'modal_a_e', company: @company %>
 
 <script>
-    let name_modal = document.getElementById('name_modal');
-    let form_modal = document.getElementById("form_modal");
-    let name_edit = document.getElementById('company_name');
-    let address_edit = document.getElementById('company_address');
-    let type_method = document.getElementById('method_put');
-    let method = '<input type="hidden" name="_method" value="put" />';
-    let button = document.getElementById('buton_action_modal');
+  let name_modal = document.getElementById('name_modal');
+  let form_modal = document.getElementById("form_modal");
+  let name_edit = document.getElementById('company_name');
+  let address_edit = document.getElementById('company_address');
+  let type_method = document.getElementById('method_put');
+  let method = '<input type="hidden" name="_method" value="put" />';
+  let button = document.getElementById('buton_action_modal');
 
-    function add()
-    {
-        name_modal.innerHTML= 'Add Company';
-        form_modal.action = '/companies'
-        name_edit.value = "";
-        address_edit.value = "";
-        type_method.innerHTML ="";
-        button.classList.remove('d-none');
-    }
+  function add(){
+    name_modal.innerHTML= 'Add Company';
+    form_modal.action = '/companies'
+    name_edit.value = "";
+    address_edit.value = "";
+    type_method.innerHTML ="";
+    button.classList.remove('d-none');
+  }
 
-    function edit(id,name,address)
-    {
-        name_modal.innerHTML= 'Edit Company';
-        form_modal.action = '/companies/'+id;
-        name_edit.value = name;
-        address_edit.value = address;
-        type_method.innerHTML =method;
-        button.classList.remove('d-none');
-    }
+  function edit(id,name,address){
+    name_modal.innerHTML= 'Edit Company';
+    form_modal.action = '/companies/'+id;
+    name_edit.value = name;
+    address_edit.value = address;
+    type_method.innerHTML =method;
+    button.classList.remove('d-none');
+  }
 
-    function show(name,address)
-    {
-        name_modal.innerHTML= 'Show Company';
-        form_modal.action = ''
-        name_edit.value = name;
-        address_edit.value = address;
-        type_method.innerHTML ='';
-        button.classList.add('d-none');
-    }
-</script>
\ No newline at end of file
+  function show(name,address){
+    name_modal.innerHTML= 'Show Company';
+    form_modal.action = ''
+    name_edit.value = name;
+    address_edit.value = address;
+    type_method.innerHTML ='';
+    button.classList.add('d-none');
+  }
+</script>

```

Comment:
This JS code, it is recommended to move it to a JS file, to call him you can make use of the [content_for Method](https://guides.rubyonrails.org/layouts_and_rendering.html#using-the-content-for-method), placing it at the top of this file.

Example:

```
<% content_for :head do %>
  <%= javascript_pack_tag 'JS_file_name' %>
<% end %>
```

Inside the JS file, you can put all this code inside addEventListener that calls the [turbolinks:load method](https://guides.rubyonrails.org/working_with_javascript_in_rails.html#turbolinks).

Example:

```
document.addEventListener('turbolinks:load', function () {
  // Your code
};
```

Reply:
ok,I'll try

Reply:
@YilKanda Webpack is creating a js file wrapping the code in a function. The functions can't be called by them name in this case, so we are saving the functions in global variables. Is there another way to do this?

![image](https://user-images.githubusercontent.com/41593304/126882875-c9a0bce3-d936-475b-875f-1a74eeb93d42.png)

Reply:
@YilKanda can you provide us an example?